tests/integration/integration-test.py

```diff
@@ -44,48 +46,60 @@ def test_get_pets_with_filter(self, endpoint='/pets'):
 
     # Test case: GET /pets with pagination parameters
     def test_get_pets_pagination(self, endpoint='/pets'):
-        testing_paginations = [
+        testing_valid_paginations = [
             {'number': 1, 'size': 25, 'expected_status_code': 200},
             {'number': 1, 'size': None, 'expected_status_code': 200},
             {'number': None, 'size': 25, 'expected_status_code': 200},
-            {'number': 999, 'size': 1, 'expected_status_code': 200},
+            {'number': 999, 'size': 1, 'expected_status_code': 200}
+        ]
+
+        testing_invalid_paginations = [
             {'number': -1, 'size': 25, 'expected_status_code': 400},
             {'number': 1, 'size': -1, 'expected_status_code': 400},
             {'number': 1, 'size': 501, 'expected_status_code': 400}
         ]
 
-        for pagination in testing_paginations:
+        for pagination in testing_valid_paginations:
             params = {f'page[{k}]': pagination[k] for k in ['number', 'size']}
             expected_status_code = pagination['expected_status_code']
-            response = utils.make_request(self, endpoint, expected_status_code,
-                                          params=params)
+            response = utils.test_endpoint(self, endpoint,
+                                           resource='PetResource',
+                                           response_code=expected_status_code,
+                                           query_params=params)
             content = utils.get_json_content(self, response)
+            try:
+                meta = content['meta']
+                num = pagination['number'] if pagination['number'] else 1
+                size = pagination['size'] if pagination['size'] else 25
 
-            if expected_status_code == 200:
-                try:
-                    meta = content['meta']
-                    num = pagination['number'] if pagination['number'] else 1
-                    size = pagination['size'] if pagination['size'] else 25
+                self.assertEqual(num, meta['currentPageNumber'])
+                self.assertEqual(size, meta['currentPageSize'])
+            except KeyError as error:
+                self.fail(error)
 
-                    self.assertEqual(num, meta['currentPageNumber'])
-                    self.assertEqual(size, meta['currentPageSize'])
-                except KeyError as error:
-                    self.fail(error)
+        for pagination in testing_invalid_paginations:
+            params = {f'page[{k}]': pagination[k] for k in ['number', 'size']}
+            expected_status_code = pagination['expected_status_code']
+            response = utils.test_endpoint(self, endpoint,
+                                           resource='Error',
+                                           response_code=expected_status_code,
+                                           query_params=params)
+            content = utils.get_json_content(self, response)
```

Comment:
The logic of line 80-87 is repeated with the line 62-68. I can tell you would like to separate valid and invalid pagination tests, but there is no additional testing between these two besides the pagination check. In this case, I'm not sure is it a good idea to separate them. What do you think?